src/content/posts/whats-possible-with-melange.mdx

```diff
@@ -0,0 +1,202 @@
+
+## Editor integration
+
+Using the latest versions of the OCaml compiler allows to use the OCaml LSP Server from the OCaml Platform. It's mature, well maintained and has better support for editor features such as autocompletion, refactoring, type lenses, documentation lookup, debugging of the transformed ppx output and an ast explorer.
+
+![Screenshot of VSCode with OCaml LSP](/images/melange-editor.png)
+
+One of my favourite features are type lenses.
+
+The previous story of the LSP in BuckleScript/Reason was a tragedy. The editor integration was limited, buggy and even when ReScript started their editor was lacking the basics. Worth mentioning, further versions of ReScript has improved this massively but still very far from the OCaml LSP.
+
+A quality of life improvement is allowing to use the same editor to write both Melange code and OCaml code, removing the need to switch between different editors when working with both targets.
+
+## Dune is a great build tool
+
+Dune is the defacto build tool for OCaml development these days. In grosso modo, it’s a wrapper to all OCaml tooling such as the Batch compiler (ocamlc), the Native compiler (ocamlopt), also the Library manager (ocamlfind), etc... It abstracts away from those low level tools for users to be in a higher level to define their libraries, executables, testing suites, documentation and other pieces needed to develop a full application.
+
+- Dune allows to rebuild your project smartly, only the libraries that change
+- Which means, It would only compile your opam dependencies only once (from the same switch)
+- Dune wouldn’t build the code where the interface didn’t change
+- Dune also builds parallely as much as possible
+- Integrates with the rest of the OCaml ecosystem, for example ppxlib or menhir.
+
+It also has been expanded to support other platforms than just native OCaml programs such as [js_of_ocaml](https://github.com/ocsigen/js_of_ocaml), [coq](https://coq.inria.fr/), [to binding C libraries](https://michael.bacarella.com/2022/02/19/dune-ctypes/), [unikernels with MirageOS](https://mirage.io/) and now Melange.
+
+Most the features that I give credit to Melange, it's thanks to dune. Will try to list the ones I appreciate the most.
+
+## Dune gives control to organize your libraries as you wish
+
+The [library stanza](https://dune.readthedocs.io/en/stable/dune-files.html#library) is useful to split your application into smaller modules and create a better organisation of your codebase. Those libraries can be defined with their own set of dependencies and ppx’s. This is extremely useful for any application older than 1 month and almost a necessity for a monorepo or big application.
```

Comment:
changing "modules" to "pieces" to avoid overloading the term

```suggestion
The [library stanza](https://dune.readthedocs.io/en/stable/dune-files.html#library) is useful to split your application into smaller pieces and create a better organisation of your codebase. Those libraries can be defined with their own set of dependencies and ppx’s. This is extremely useful for any application older than 1 month and almost a necessity for a monorepo or big application.
```